app/javascript/components/Discover/Layout.tsx

```diff
@@ -21,16 +21,32 @@ export const Layout: React.FC<{
   setQuery?: (query: string) => void;
   className?: string;
   children: React.ReactNode;
-}> = ({ taxonomiesForNav, taxonomyPath, showTaxonomy, onTaxonomyChange, query, setQuery, className, children }) => {
+  forceDomain?: boolean;
+}> = ({
+  taxonomiesForNav,
+  taxonomyPath,
+  showTaxonomy,
+  onTaxonomyChange,
+  query,
+  setQuery,
+  className,
+  children,
+  forceDomain = false,
+}) => {
   const { discoverDomain } = useDomains();
   const isDesktop = useIsAboveBreakpoint("lg");
   const currentSeller = useCurrentSeller();
 
   const rootTaxonomy = getRootTaxonomy(taxonomyPath);
 
   setQuery ??= (query) => (window.location.href = Routes.discover_url({ host: discoverDomain, query }));
-  onTaxonomyChange ??= (newTaxonomyPath) =>
-    (window.location.href = Routes.discover_url({ host: discoverDomain, taxonomy: newTaxonomyPath }));
+  onTaxonomyChange ??= (newTaxonomyPath) => (window.location.href = newTaxonomyPath ? newTaxonomyPath : "/");
+
+  onTaxonomyChange ??= (newTaxonomyPath) => {
+    window.location.href = forceDomain
+      ? newTaxonomyPath || "/"
+      : Routes.discover_url({ host: discoverDomain, taxonomy: newTaxonomyPath });
+  };
```

Comment:
Duplicate assignment, second assignment will be always skipped?

Reply:
Hmm no, because if the `forceDomain` is true, then the `newTaxonomyPath` already has the domain.
if `forceDomain` is false, then we need to append the domain.